nixos/modules/hardware/antec.nix

```diff
@@ -0,0 +1,84 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.modules) mkIf;
+  inherit (lib.meta) getExe;
+  inherit (lib.types) str;
+  inherit (lib.options) mkEnableOption mkOption;
+
+  antec-flux-pro-pkg = pkgs.antec-flux-pro.override {
+    cpu_device = cfg.cpu-device;
+    cpu_temp_type = cfg.cpu-temp-type;
+    gpu_device = cfg.gpu-device;
+    gpu_temp_type = cfg.gpu-temp-type;
+  };
+  cfg = config.hardware.antec;
+in
+{
+  meta.maintainers = [ lib.maintainers.kruziikrel13 ];
+
+  options = {
+    hardware.antec = {
+      enable = mkEnableOption "Enable support for Antec Flux Pro GPU and CPU Temperature Sensor.";
+      cpu-device = mkOption {
+        type = str;
+        default = "k10temp";
+        description = "CPU temperature device name";
+      };
+      cpu-temp-type = mkOption {
+        type = str;
+        default = "tctl";
+        description = "CPU temperature sensor label";
+      };
+      gpu-device = mkOption {
+        type = str;
+        default = "amdgpu";
+        description = "GPU temperature device name";
+      };
+      gpu-temp-type = mkOption {
+        type = str;
+        default = "edge";
+        description = "GPU temperature sensor label";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ antec-flux-pro-pkg ];
+    services.udev.packages = [ antec-flux-pro-pkg.udev ];
+
+    systemd.services.antec-flux-pro-display = {
+      unitConfig = {
+        Description = "Antec Flux Pro Display Service";
+        StartLimitIntervalSec = "5";
+      };
+
+      serviceConfig = {
+        Type = "simple";
+        ExecStartPre = "${pkgs.coreutils}/bin/sleep 5";
+        ExecStart = getExe antec-flux-pro-pkg;
+        Restart = "always";
+        RestartSec = "5";
+        ProtectSystem = "strict";
+        ProtectHome = "read-only";
+        PrivateTmp = "true";
+        NoNewPrivileges = "true";
+        MemoryDenyWriteExecute = "true";
+        RestrictRealtime = "true";
+
+        LimitNOFILE = "1024";
+        MemoryMax = "128M";
```

Comment:
You are absolutely certain this service will never use more than 128M, and if it does it's a memory leak?